src/hotspot/share/runtime/flags/allFlags.hpp

```diff
@@ -0,0 +1,120 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_RUNTIME_FLAGS_ALLFLAGS_HPP
+#define SHARE_RUNTIME_FLAGS_ALLFLAGS_HPP
+
+#include "compiler/compiler_globals.hpp"
+#include "runtime/globals.hpp"
+
+// Put the LP64/JVMCI/COMPILER1/COMPILER1/ARCH at
+// the top, as they are processed by jvmFlags.cpp in that
+// order.
+
+#define ALL_FLAGS(            \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint)               \
+                              \
+  LP64_RUNTIME_FLAGS(         \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint)               \
+                              \
+  JVMCI_ONLY(JVMCI_FLAGS(     \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint))              \
+                              \
+  COMPILER1_PRESENT(C1_FLAGS( \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint))              \
+                              \
+  COMPILER2_PRESENT(C2_FLAGS( \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint))              \
+                              \
+  ARCH_FLAGS(                 \
+    develop,                  \
+    product,                  \
+    notproduct,               \
+    range,                    \
+    constraint)               \
+                              \
+  RUNTIME_FLAGS(              \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint)               \
+                              \
+  GC_FLAGS(                   \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint)               \
+                              \
+  RUNTIME_OS_FLAGS(           \
+    develop,                  \
+    develop_pd,               \
+    product,                  \
+    product_pd,               \
+    notproduct,               \
+    range,                    \
+    constraint)
```

Comment:
RUNTIME / GC / RUNTIME_OS : Could the order be changed so that the RUNTIME macros are grouped together?